daft/filesystem.py

```diff
@@ -353,3 +358,22 @@ def join_path(fs: pafs.FileSystem, base_path: str, *sub_paths: str) -> str:
         return os.path.join(base_path, *sub_paths)
     else:
         return f"{base_path.rstrip('/')}/{'/'.join(sub_paths)}"
+
+
+def overwrite_files(
```

Comment:
It doesn't look like we're supporting partition overwrite mode here. Where we only delete files if we wrote a new file into that partition.

Could we leverage what @desmondcheongzx is working on for hive style reads to be able to do this?

Reply:
Probably, but we could also just match on the directory paths and delete accordingly. Will make a separate PR for this.